### PR TITLE
VxAdmin: Polish manual tally entry flow

### LINKED ISSUE #5295: VxAdmin: Manual results, prompt to enter number of votes for write-in candidate after adding
When a write-in is added to the manual tallies, it would be helpful if there was a prompt to enter the number of votes for that write-in candidate. It defaults to zero now and New Durham missed entering the count a few times. It wouldn’t have been caught if they didn’t use the total number of ballots in each race box. 

Issue automatically created from Slack feedback reported by ginny: 
https://votingworks.slack.com/archives/CJU9MSC6S/p1723993921236839

### LINKED ISSUE #5381: VxAdmin: manual results, green checkmark at bottom of contest for long list of write-ins
We have validation on each contest (green check marks) but they appear above the contests and, for a long contest, it may be best to have it below instead or in addition to the top. 

Issue automatically created from Slack feedback reported by ben: 
https://votingworks.slack.com/archives/CJU9MSC6S/p1726024880787059

### LINKED ISSUE #5390: VxAdmin: Stricter form completion logic on manual results
Instead of only using validation symbols on manual results, we could just prevent saving results until the results are valid.

Also, it's very easy to exit the form and lose a lot of work - we should have a confirmation modal there if any results have been edited.

### LINKED ISSUE #5407: Customer mentioned that for manual tally input, it would be nice to have a "ballots cast - apply to all" type choice so that for each race we don't have to input the # of ballots, if we so choose
Customer mentioned that for manual tally input, it would be nice to have a "ballots cast - apply to all" type choice so that for each race we don't have to input the # of ballots, if we so choose

Issue automatically created from Slack feedback reported by janine: 
https://votingworks.slack.com/archives/CJU9MSC6S/p1726248143702439

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Reworks the manual tally entry flow to address a number of UX issues. The main changes:
- Enter and save tallies for one contest at a time 
  Fixes #5390 
  Fixes #5381

- Default tallies to blank, not 0, and disallow saving until you fill out all the tallies 
  Fixes #5295
- Enter an overall ballot count once at the beginning, then allow overriding for a specific contest if needed
  Fixes #5407

## Demo Video or Screenshot

https://github.com/user-attachments/assets/7183bb99-bee3-40b5-95e7-288d4de3a036

## Testing Plan
Manual testing, updated automated tests

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx [hunk 1]
```diff
@@ -554,7 +593,10 @@ function ManualResultsDataEntryScreenForm({
                 variant="primary"
                 icon="Done"
                 onPress={saveResults}
-                disabled={setManualTallyMutation.isLoading}
+                disabled={
+                  someContestHasIncompleteResults ||
```
Comment (nbhatia823) [inline apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx]: No need to check as well for `someContestHasEmptyResults`?

File: apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx [hunk 2]
```diff
@@ -589,129 +755,168 @@ function ManualResultsDataEntryScreenForm({
...
-                addWriteInCandidate={(name) => addFormWriteInCandidate(name)}
-                contestId={contest.id}
-                disallowedCandidateNames={disallowedNewWriteInCandidateNames}
-              />
-            )}
+                </React.Fragment>
+              )}
+              {contest.type === 'candidate' && contest.allowWriteIns && (
+                <AddWriteInRow
+                  addWriteInCandidate={(name) => addFormWriteInCandidate(name)}
```
Comment (nbhatia823) [inline apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx]: nit: style option of just making it `addWriteInCandidate={addFormWriteInCandidate}`

**Review by nbhatia823 — APPROVED** [review]
Reviewed commit by commit and LGTM!

File: apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx (hunk 1 above)
Comment (jonahkagan) [reply apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx]: I think this code gets removed later, so it's not relevant, is that correct?

File: apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx (hunk 2 above)
Comment (jonahkagan) [reply apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx]: 3e9944974

File: apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx (hunk 1 above)
Comment (nbhatia823) [reply apps/admin/frontend/src/screens/tally/manual_tallies_form_screen.tsx]: Oop right, sorry about that. Commit by commit review issue on my end

**Pull request merged**